mmv1/products/tags/api.yaml

```diff
@@ -0,0 +1,99 @@
+# Copyright 2021 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: Tags
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://cloudresourcemanager.googleapis.com/v3/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+objects:
+  - !ruby/object:Api::Resource
+    name: 'TagKey'
+    min_version: beta
+    base_url: tagKeys
+    self_link: "{{name}}"
```

Comment:
Can we set the self_link instead to "tagKeys/{{name}}" and parse out the tag_key_id from the API response to store the value individually? I think it makes things more usable when we split out a longer form name into the individual parts.

Then we can set the import formats to accept either `tagKeys/{{name}}` or `{{name}}` directly.

Reply:
`name_from_self_link` custom flatten can help parse this

Reply:
I had originally tried having that as the self_link and was running into issues... but I didn't have it combined with the custom flatten. I'll give it a shot. +1 on the general concept :-)